src/addPropsOnChange.coffee

```diff
@@ -1,24 +1,6 @@
 import addProps from './addProps'
-import usePrevious from './util/usePrevious'
-import {isArray} from './util/helpers'
 
 addPropsOnChange = (didChange, getProps) ->
-  return addProps getProps, didChange if isArray didChange
-
-  prevAddedProps = null
```

Comment:
The existing implementation of manual "memoization" in `addPropsOnChange()` (for tracking the last computed value when supplied with a callback dependencies argument) used a single "global" variable for tracking the last computed value, which would've been incorrect with respect to multiple instances of the same component

So the new shared `useComputedFromDependencies()` helper uses `useRef()` instead which should give each instance of a component its own "history" of previously computed values